### PR TITLE
content avail service was added

### PR DESCRIPTION
**Pull request by igormas**

### CODE REVIEW

File: dist/znk.infra.js [hunk 1]
```diff
@@ -240,6 +247,138 @@
...
+                var freeContentPath = 'freeContent';
+                var defValues = {
+                    daily: 0,
+                    exam: {},
+                    tutorial: {}
+                };
+                return StorageService.get(freeContentPath,defValues);
+            }
+
+            function idToKeyInStorage(id){
```
Comment (assafshp) [inline dist/znk.infra.js]: WHY DO WE NEED IT?

File: dist/znk.infra.js [hunk 2]
```diff
@@ -240,6 +247,138 @@
...
+            this.isDailyAvail = function(dailyOrder){
+                return _baseIsEntityAvail().then(function(res){
+                    if(res === true){
+                        return true;
+                    }
+
+                    var purchaseData = res[0];
+                    var freeContent = res[1];
+
+                    if(freeContent.daily >= dailyOrder){
```
Comment (assafshp) [inline dist/znk.infra.js]: MAYBE you need to check if dailyorder is number

File: dist/znk.infra.js [hunk 3]
```diff
@@ -240,6 +247,138 @@
...
+
+            this.hasSubscription = function(){
+                return getUserPurchaseData().then(function(purchaseData){
+                    return _hasSubscription(purchaseData.subscription);
+                });
+            };
+
+            this.isDailyAvail = function(dailyOrder){
+                return _baseIsEntityAvail().then(function(res){
+                    if(res === true){
```
Comment (assafshp) [inline dist/znk.infra.js]: can't this part be also in the _base becasue it is duplicate in all usage of _base

File: dist/znk.infra.js [hunk 4]
```diff
@@ -240,6 +247,138 @@
...
+            function idToKeyInStorage(id){
+                return 'id_' + id;
+            }
+
+            function _hasSubscription(subscriptionObj){
+                return subscriptionObj && subscriptionObj.expiryDate && subscriptionObj.expiryDate > Date.now();
+            }
+
+            function _baseIsEntityAvail(){
+                return $q.all([getUserPurchaseData(),getFreeContentData()]).then(function(res){
```
Comment (assafshp) [inline dist/znk.infra.js]: [getUserPurchaseData(),getFreeContentData()] are cached ? should be

File: test/mock/storage/module.js [hunk 5]
```diff
@@ -47,6 +45,8 @@
             storage.db = db;
             storage.variables = StorageSrv.variables;
 
+            setInDb(storage.variables.appUserSpacePath,{});
```
Comment (assafshp) [inline test/mock/storage/module.js]: why ?

File: test/spec/contentAvail/contentAvailSrv.spec.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    });
+
+    it('given user has subscription when asking isExamAvail then should return true for all exam ids',function(){
+        expect(actions.isExamAvail(3)).toBeFalsy();
+        actions.addSubscription();
+        expect(actions.isExamAvail(3)).toBeTruthy();
+    });
+
+    it('given user purchased daily bundle and free daily is 1 when checking if daily available then only purchased or free dailies should be available',function(){
+        actions.setFreeDaily(1);
```
Comment (assafshp) [inline test/spec/contentAvail/contentAvailSrv.spec.js]: maybe add assestion after this line
expect(actions.isDailyAvail(2)).toBeFalsy();

File: dist/znk.infra.js (hunk 1 above)
Comment (igormas) [inline dist/znk.infra.js]: If we attach without the 'id_' prefix then firebase will treat it as array.

File: dist/znk.infra.js (hunk 4 above)
Comment (igormas) [inline dist/znk.infra.js]: They will be cached by storage srv

File: dist/znk.infra.js (hunk 3 above)
Comment (igormas) [inline dist/znk.infra.js]: No ....

File: dist/znk.infra.js (hunk 2 above)
Comment (igormas) [inline dist/znk.infra.js]: Fixed

File: test/mock/storage/module.js (hunk 5 above)
Comment (igormas) [inline test/mock/storage/module.js]: Its storage mock, just setting the initial values.

File: test/spec/contentAvail/contentAvailSrv.spec.js (hunk 6 above)
Comment (igormas) [inline test/spec/contentAvail/contentAvailSrv.spec.js]: Fixed

**Pull request merged**